debug-ptrace: gcc8 warnings

```
[925/1124] Compiling C object 'libr/debug/libr@debug@@r_debug@sha/p_debug_native.c.o'.                                                                        
../libr/debug/p/debug_native.c: In function ‘r_debug_native_continue’: 
../libr/debug/p/debug_native.c:276:57: warning: passing argument 5 of ‘r_debug_ptrace’ makes pointer from integer without a cast [-Wint-conversion]
  int ret = r_debug_ptrace (dbg, PTRACE_CONT, pid, NULL, contsig);
                                                         ^~~~~~~
In file included from ../libr/debug/p/debug_native.c:4:            
../libr/include/r_debug.h:603:115: note: expected ‘r_ptrace_data_t’ {aka ‘void *’} but argument is of type ‘int’
 static inline long r_debug_ptrace(RDebug *dbg, r_ptrace_request_t request, pid_t pid, void *addr, r_ptrace_data_t data) {
                                                                                                   ~~~~~~~~~~~~~~~~^~~~
../libr/debug/p/debug_native.c:288:51: warning: passing argument 5 of ‘r_debug_ptrace’ makes pointer from integer without a cast [-Wint-conversion]
      r_debug_ptrace (dbg, PTRACE_CONT, tid, NULL, contsig);               
                                                   ^~~~~~~
In file included from ../libr/debug/p/debug_native.c:4:
../libr/include/r_debug.h:603:115: note: expected ‘r_ptrace_data_t’ {aka ‘void *’} but argument is of type ‘int’
 static inline long r_debug_ptrace(RDebug *dbg, r_ptrace_request_t request, pid_t pid, void *addr, r_ptrace_data_t data) {
                                                                                                   ~~~~~~~~~~~~~~~~^~~~
[930/1124] Compiling C object 'libr/debug/libr@debug@@r_debug@sha/p_native_linux_linux_debug.c.o'.
../libr/debug/p/native/linux/linux_debug.c: In function ‘linux_handle_signals’:
../libr/debug/p/native/linux/linux_debug.c:81:2: warning: #warning DO MORE RDEBUGREASON HERE [-Wcpp]
 #warning DO MORE RDEBUGREASON HERE
  ^~~~~~~
../libr/debug/p/native/linux/linux_debug.c: In function ‘linux_set_options’:
../libr/debug/p/native/linux/linux_debug.c:242:54: warning: passing argument 5 of ‘r_debug_ptrace’ makes pointer from integer without a cast [-Wint-conversio$
]
  if (r_debug_ptrace (dbg, PTRACE_SETOPTIONS, pid, 0, traceflags) == -1) {
                                                      ^~~~~~~~~~
In file included from ../libr/debug/p/native/linux/linux_debug.c:6:
../libr/include/r_debug.h:603:115: note: expected ‘r_ptrace_data_t’ {aka ‘void *’} but argument is of type ‘int’
 static inline long r_debug_ptrace(RDebug *dbg, r_ptrace_request_t request, pid_t pid, void *addr, r_ptrace_data_t data) {
                                                                                                   ~~~~~~~~~~~~~~~~^~~~
In file included from ../libr/include/sdb.h:1,
                 from ../libr/include/r_list.h:6,
                 from ../libr/include/r_io.h:6,
                 from ../libr/include/r_anal.h:16,
                 from ../libr/include/r_debug.h:5,
                 from ../libr/debug/p/native/linux/linux_debug.c:6:
../libr/debug/p/native/linux/linux_debug.c: In function ‘linux_reg_read’:
../libr/include/sdb/sdb.h:21:35: warning: passing argument 4 of ‘r_debug_ptrace’ makes pointer from integer without a cast [-Wint-conversion]
 #define r_offsetof(type, member) ((unsigned long) &((type*)0)->member)
                                 ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../libr/debug/p/native/linux/linux_debug.c:790:6: note: in expansion of macro ‘r_offsetof’
      r_offsetof (struct user, u_debugreg[i]), 0);
      ^~~~~~~~~~
In file included from ../libr/debug/p/native/linux/linux_debug.c:6:
../libr/include/r_debug.h:603:93: note: expected ‘void *’ but argument is of type ‘long unsigned int’
 static inline long r_debug_ptrace(RDebug *dbg, r_ptrace_request_t request, pid_t pid, void *addr, r_ptrace_data_t data) {
                                                                                       ~~~~~~^~~~
In file included from ../libr/include/sdb.h:1,
                 from ../libr/include/r_list.h:6,
                 from ../libr/include/r_io.h:6,
                 from ../libr/include/r_anal.h:16,
                 from ../libr/include/r_debug.h:5,
                 from ../libr/debug/p/native/linux/linux_debug.c:6:
../libr/debug/p/native/linux/linux_debug.c: In function ‘linux_reg_write’:
../libr/include/sdb/sdb.h:21:35: warning: passing argument 4 of ‘r_debug_ptrace’ makes pointer from integer without a cast [-Wint-conversion]
 #define r_offsetof(type, member) ((unsigned long) &((type*)0)->member)
                                  ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../libr/debug/p/native/linux/linux_debug.c:921:56: note: in expansion of macro ‘r_offsetof’
    if (r_debug_ptrace (dbg, PTRACE_POKEUSER, dbg->pid, r_offsetof (
                                                        ^~~~~~~~~~
In file included from ../libr/debug/p/native/linux/linux_debug.c:6:
../libr/include/r_debug.h:603:93: note: expected ‘void *’ but argument is of type ‘long unsigned int’
 static inline long r_debug_ptrace(RDebug *dbg, r_ptrace_request_t request, pid_t pid, void *addr, r_ptrace_data_t data) {
                                                                                       ~~~~~~^~~~
../libr/debug/p/native/linux/linux_debug.c:922:38: warning: passing argument 5 of ‘r_debug_ptrace’ makes pointer from integer without a cast [-Wint-conversion
]
      struct user, u_debugreg[i]), val[i])) {
                                   ~~~^~~
In file included from ../libr/debug/p/native/linux/linux_debug.c:6:
../libr/include/r_debug.h:603:115: note: expected ‘r_ptrace_data_t’ {aka ‘void *’} but argument is of type ‘long int’
 static inline long r_debug_ptrace(RDebug *dbg, r_ptrace_request_t request, pid_t pid, void *addr, r_ptrace_data_t data) {
                                                                                                   ~~~~~~~~~~~~~~~~^~~~
[943/1124] Compiling C object 'libr/core/libr@core@@r_core@sha/cfile.c.o'.
../libr/core/cfile.c: In function ‘r_core_file_reopen’:
../libr/core/cfile.c:117:9: note: #pragma message: fix debugger-concept in core
 #pragma message ("fix debugger-concept in core")
         ^~~~~~~

--- a/libr/debug/p/debug_native.c
+++ b/libr/debug/p/debug_native.c
@@ -273,7 +273,7 @@ static int r_debug_native_continue(RDebug *dbg, int pid, int tid, int sig) {
 		r_cons_break_push ((RConsBreak)r_debug_native_stop, dbg);
 	}
 
-	int ret = r_debug_ptrace (dbg, PTRACE_CONT, pid, NULL, contsig);
+	int ret = r_debug_ptrace (dbg, PTRACE_CONT, pid, NULL, (r_ptrace_data_t)(size_t)contsig);
 	if (ret) {
 		perror ("PTRACE_CONT");
 	}
@@ -285,7 +285,7 @@ static int r_debug_native_continue(RDebug *dbg, int pid, int tid, int sig) {
 		if (list) {
 			r_list_foreach (list, it, th) {
 				if (th->pid && th->pid != pid) {
-					r_debug_ptrace (dbg, PTRACE_CONT, tid, NULL, contsig);
+					r_debug_ptrace (dbg, PTRACE_CONT, tid, NULL, (r_ptrace_data_t)(size_t)contsig);
 				}
 			}
 		}

--- a/libr/debug/p/native/linux/linux_debug.c
+++ b/libr/debug/p/native/linux/linux_debug.c
@@ -239,7 +239,7 @@ bool linux_set_options(RDebug *dbg, int pid) {
 	}
 	/* SIGTRAP | 0x80 on signal handler .. not supported on all archs */
 	traceflags |= PTRACE_O_TRACESYSGOOD;
-	if (r_debug_ptrace (dbg, PTRACE_SETOPTIONS, pid, 0, traceflags) == -1) {
+	if (r_debug_ptrace (dbg, PTRACE_SETOPTIONS, pid, 0, (r_ptrace_data_t)(size_t)traceflags) == -1) {
 		return false;
 	}
 	return true;
@@ -787,7 +787,7 @@ int linux_reg_read (RDebug *dbg, int type, ut8 *buf, int size) {
 				continue;
 			}
 			long ret = r_debug_ptrace (dbg, PTRACE_PEEKUSER, pid,
-					r_offsetof (struct user, u_debugreg[i]), 0);
+					(void *)r_offsetof (struct user, u_debugreg[i]), 0);
 			if ((i+1) * sizeof (ret) > size) {
 				eprintf ("linux_reg_get: Buffer too small %d\n", size);
 				break;
@@ -918,8 +918,8 @@ int linux_reg_write (RDebug *dbg, int type, const ut8 *buf, int size) {
 			if (i == 4 || i == 5) {
 				continue;
 			}
-			if (r_debug_ptrace (dbg, PTRACE_POKEUSER, dbg->pid, r_offsetof (
-					struct user, u_debugreg[i]), val[i])) {
+			if (r_debug_ptrace (dbg, PTRACE_POKEUSER, dbg->pid,
+					(void *)r_offsetof (struct user, u_debugreg[i]), (r_ptrace_data_t)val[i])) {
 				eprintf ("ptrace error for dr %d\n", i);
 				r_sys_perror ("ptrace POKEUSER");
 			}

